core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/execution/ExecutionUtils.scala

```diff
@@ -0,0 +1,70 @@
+package edu.uci.ics.amber.engine.architecture.controller.execution
+
+import edu.uci.ics.texera.web.workflowruntimestate.{OperatorRuntimeStats, WorkflowAggregatedState}
+
+object ExecutionUtils {
+
+  /**
+    * Handle the case when a logical operator has two physical operators within a same region (e.g., Aggregate operator)
+    */
+  def aggregateStats(stats: Iterable[OperatorRuntimeStats]): OperatorRuntimeStats = {
+    val aggregatedState = aggregateStates(
+      stats.map(_.state.value),
+      WorkflowAggregatedState.COMPLETED.value,
+      WorkflowAggregatedState.RUNNING.value,
+      WorkflowAggregatedState.UNINITIALIZED.value,
+      WorkflowAggregatedState.PAUSED.value,
+      WorkflowAggregatedState.READY.value
+    )
+
+    val inputCountSum = stats.flatMap(_.inputCount).groupBy(_._1).map {
+      case (k, v) =>
+        k -> v.map(_._2).sum
+    }
+    val outputCountSum = stats.flatMap(_.outputCount).groupBy(_._1).map {
+      case (k, v) =>
+        k -> v.map(_._2).sum
+    }
```

Comment:
I just want to make sure you are filtering out the internal ports for calculation. If the input of this function is already assumed to be only containing external ports, please reflect it in the scala-doc of this function.

Reply:
I'll fix this in the different PR